Marathon Cloud doc update (#807)
MarathonLabs/marathon: docs/docs/cloud.md

---

> title: "Overview"

Reviewer: Why is this deleted?

# Introduction

[Marathon Cloud](marathonlabs.io) revolutionizes your app testing experience. The platform provides infinite virtual devices and will automatically shard, sort, distribute, and retry your tests, allowing all tests to be completed in **a maximum of 15 minutes**. First **50 hours** are free!

## Getting started

### Install

> The installation can be performed using [Homebrew](https://brew.sh/). Here's how to add the MarathonLabs repository:

Reviewer: You can use `brew install malinskiy/tap/marathon-cloud`. It adds the tap automatically

```shell
brew tap malinskiy/tap
```
Next, install the Marathon Cloud CLI:
```shell
brew install malinskiy/tap/marathon-cloud
```
Alternatively, you can download prebuilt binaries for Windows, Linux, or MacOS from [the Release page](https://github.com/MarathonLabs/marathon-cloud-cli/releases).

### API Key

> Token creation and management are available at [the Tokens page](https://dev.testwise.pro/tokens). Generate a token and save it somewhere safe for the next step.

Reviewer: Link should be to the public cloud

The test execution is then delegated to Marathon test runner, which handles tasks such as batching, 
sorting, preventive retries, and post-factum retries. 
This ensures an even distribution of tests across the provisioned devices. 

## Install

The installation can be performed using [Homebrew](https://brew.sh/):
```shell
brew install malinskiy/tap/marathon-cloud
```
Alternatively, you can download prebuilt binaries for Windows, Linux, or MacOS from [the Release page](https://github.com/MarathonLabs/marathon-cloud-cli/releases).

## API Key

Token creation and management are available at [the Tokens page](https://cloud.marathonlabs.io/tokens). Generate a token and save it somewhere safe for the next step.

## Prepare bundles for testing

### iOS

Marathon Cloud supports tests written with **XCTest and XCUITest frameworks**.
Both the application and the tests must be built for the **ARM architecture**.

Before initiating the testing process for your iOS application, you’ll need to create two `.app` bundles: one for the application that's being tested, and another for the tests themselves. Typically, `debug` variants are utilized for this purpose.

> Imagine that our project is called "Sample". For this scenario, the code snippet below demonstrates how the building command should be presented:

Reviewer: ```suggestion
Let's say our project is called "Sample". The code snippet below shows how to build the .app bundle:
```

```shell
# file structure
# |
# |--home
#    |--john
#       |--sample <== you are here
#          |--sample  <== it's your application
#          ...
#          |--sample.xcodeproj

xcodebuild build-for-testing \
  -project sample.xcodeproj \
  -scheme sample \
  -destination 'platform=iOS Simulator,name=iPhone 14,OS=16.1' \
  -derivedDataPath ./build
```

> Be sure to note the relative paths of applications, as they will be required for running the tests. In the context of our example, involving the `debug` build, these files can be located at the following paths:

Reviewer: ```suggestion
Note the relative paths of applications, as they will be required for running the tests. In the context of our example and `debug` build, these files can be located at the following paths:
```

- Application: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sample.app`
- Test APK: `/home/john/sample/build/Build/Products/Debug-iphonesimulator/sampleUITests-Runner.app`

> One important thing to note is that `*.app` files are actually folders in disguise. In order to transfer them, it's necessary to convert these bundles into standard zip archives:

Reviewer: Please mention that ipa bundles are also supported

Reviewer: Also "In order to" - > "To"

Before initiating the testing process for your application, you’ll require two APK files: one for the application that’s being tested, and another for the tests themselves. Typically, `debug` variants are utilized for this purpose.

If the primary application resides under the `app/` subproject, you can execute the following command to build both the app and test APK:

```shell
# file structure
# |
# |--home
#    |--john
#       |--project <== you are here
#          |--app  <== it's your primary application
#          ...
#          |--build.gragle
#          |--settings.gradle  
./gradlew :app:assembleDebug :app:assembleDebugAndroidTest
```

Be sure to note the relative paths of the test APK and the app APK, as they will be required for running the tests. In the context of our example, involving the `app` project and the `debug` build, these files can be located at the following paths:

- App APK: `/home/john/project/app/build/outputs/apk/debug/app-debug.apk`
- Test APK: `/home/john/project/app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk`

## Samples (optional)

> To showcase the advantages of Marathon Cloud compared to other solutions, we've prepared a sample app with 300 tests, out of which 15% are flaky. During the initial run, our platform will gather information about the tests. During the second run, it will optimize it to fit within 15 minutes.

Reviewer: ```suggestion
To showcase the advantages of Marathon Cloud compared to other solutions, we've prepared a sample app with 300 tests, of which 15% are flaky. During the initial run, our platform will gather information about the tests. During the second run, it will optimize it to fit within 15 minutes.
```
